src/ccnl-riot/include/ccn-lite-riot.h

```diff
@@ -342,6 +342,15 @@ static inline void ccnl_riot_interest_remove(evtimer_t *et, struct ccnl_interest
 {
     evtimer_del(et, (evtimer_event_t *)&i->evtmsg_retrans);
     evtimer_del(et, (evtimer_event_t *)&i->evtmsg_timeout);
+
+    /* remove messages that relate to this interest from the message queue */
+    thread_t *me = (thread_t*) sched_threads[sched_active_pid];
```

Comment:
I wonder whether we should disable interrupts while iterating the message queue?

Reply:
Yes. I've added that and amended to the first commit.